clients/runtime/src/rpc.rs

```diff
@@ -1352,22 +1370,35 @@ impl ReplacePallet for SpacewalkParachain {
 	}
 
 	/// Get all replace requests made by the given vault
-	async fn get_old_vault_replace_requests(
+	async fn get_old_vault_replace_requests<T>(
 		&self,
 		account_id: AccountId,
-	) -> Result<Vec<(H256, SpacewalkReplaceRequest)>, Error> {
+		filter: Box<dyn Fn((H256, SpacewalkReplaceRequest)) -> Option<T> + Send>,
+	) -> Result<Vec<T>, Error> {
 		let head = self.get_finalized_block_hash().await?;
 		let result: Vec<H256> = self
 			.api
 			.rpc()
 			.request("replace_getOldVaultReplaceRequests", rpc_params![account_id, head])
 			.await?;
-		join_all(result.into_iter().map(|key| async move {
-			self.get_replace_request(key).await.map(|value| (key, value))
-		}))
-		.await
-		.into_iter()
-		.collect()
+		
+		let redeem_requests: Result<Vec<(H256, SpacewalkReplaceRequest)>, Error> =
+			join_all(result.into_iter().map(|key| async move {
+				self.get_replace_request(key).await.map(|value| (key, value))
+			}))
+			.await
+			.into_iter()
+			.collect();
+
+		match redeem_requests {
+			Ok(redeem_requests) => {
+				let filtered_results: Vec<_> =
+					redeem_requests.into_iter().filter_map(|item| filter(item)).collect();
+					Ok(filtered_results)
+				},
+			Err(e) => Err(e),
```

Comment:
Instead of collecting and turning it into an iter again, can't we somehow combine and simplify these statements?

Reply:
Sure, you are right, no real need to collect redeem_requests in the first place.